src/client/Microsoft.Identity.Client/MsalErrorMessage.cs

```diff
@@ -450,5 +450,7 @@ public static string InvalidTokenProviderResponseValue(string invalidValueName)
         public const string InvalidCertificate = "The certificate received from the Imds server is invalid.";
         public const string CannotSwitchBetweenImdsVersionsForPreview = "ImdsV2 is currently experimental - A Bearer token has already been received; Please restart the application to receive a mTLS PoP token.";
         public const string MtlsPopTokenNotSupportedinImdsV1 = "A mTLS PoP token cannot be requested because the application\'s source was determined to be ImdsV1.";
+        public const string IdentityNotFound = "Managed Identity not found. To request credentials for this identity, please assign it first. Please see aka.ms/ManagedIdentityNotFound for more details";

```

Comment:
IMDS already returns the `IdentityNotFound` error message, and MSAL simply bubbles up the IMDS payload without rewriting or overriding it. Adding our own string introduces duplication and may cause divergence from IMDS' authoritative error.

Reply:
Agreed with @gladjohn - we can depend on the error code, but the error messages should from the server.